pkg/logger/logger.go

```diff
@@ -0,0 +1,87 @@
+package logger
+
+import (
+	"os"
+	"testing"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest"
+	"go.uber.org/zap/zaptest/observer"
+)
+
+// Logger is a minimal subset of smartcontractkit/chainlink/core/logger.Logger
+type Logger interface {
+	With(args ...interface{}) Logger
+	Named(name string) Logger
+
+	Debug(args ...interface{})
+	Info(args ...interface{})
+	Warn(args ...interface{})
+	Error(args ...interface{})
+	Panic(args ...interface{})
+	Fatal(args ...interface{})
+
+	Debugf(format string, values ...interface{})
+	Infof(format string, values ...interface{})
+	Warnf(format string, values ...interface{})
+	Errorf(format string, values ...interface{})
+	Panicf(format string, values ...interface{})
+	Fatalf(format string, values ...interface{})
+
+	Debugw(msg string, keysAndValues ...interface{})
+	Infow(msg string, keysAndValues ...interface{})
+	Warnw(msg string, keysAndValues ...interface{})
+	Errorw(msg string, keysAndValues ...interface{})
+	Panicw(msg string, keysAndValues ...interface{})
+	Fatalw(msg string, keysAndValues ...interface{})
+
+	Sync() error
+}
+
+// New returns a new Logger.
+func New() (Logger, error) {
+	var lvl zapcore.Level
+	err := lvl.Set(os.Getenv("LOG_LEVEL"))
```

Comment:
I would prefer if log level was passed as a parameter to the constructor. 
I don't like env vars peppered around the code and the dependencies!
_my 2c_

Reply:
Will do! Almost did it before but was trying to capture the API exactly at first.